src/lib/caches/Items.js

```diff
@@ -94,22 +100,74 @@ module.exports = class ItemsCache {
       let exact = false;
       filtered = undefined;
       base
-        .filter((item) => item && item[by])
+        .filter((item) => item && this.#parseNestedKey(item, bys))
```

Comment:
i'm pretty sure an empty array is truthy, so you'll probably want 
```suggestion
        .filter((item) => item && this.#parseNestedKey(item, bys)?.length)
```

Reply:
parseNestedKey won't return an empty array so that check shouldn't be needed.

`return (resultList.length > 0 ? resultList : undefined);`

Reply:
then this type is incorrect:
```js
* @returns {Array<string>} or undefined if an invalid path of keys is provided
```

should either be
```js
* @returns {Array<string>?} list of results or undefined if no matching keys
```
or 
```js
* @returns {Array<string> | undefined} list of results or undefined if no matching keys
```

because "invalid" indicates a state of absoluteness, like that it could never be valid, but in many cases there's keys that won't exist on, say, skins, but would exsist on weapons